src/common/PluginRegistry.cc

```diff
@@ -93,6 +93,9 @@ int PluginRegistry::add(const std::string& type,
   }
   ldout(cct, 1) << __func__ << " " << type << " " << name
 		<< " " << plugin << dendl;
+  if (!plugins.count(type)){
+    plugins[type] = std::map<std::string,Plugin*>();
```

Comment:
i don't think this helps. `plugins[type]` creates a new element using the default constructor if `!plugins.count(type) == 0`.

could you paste the backtrace in the ticket?